macros/processor/provider.go

```diff
@@ -0,0 +1,136 @@
+package processor
+
+import (
+	"net/url"
+	"strconv"
+
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const (
+	BidIDKey          = "PBS-BIDID"
+	AppBundleKey      = "PBS-APPBUNDLE"
+	DomainKey         = "PBS-APPBUNDLE"
+	PubDomainkey      = "PBS-PUBDOMAIN"
+	PageURLKey        = "PBS-PAGEURL"
+	AccountIDKey      = "PBS-ACCOUNTID"
+	LmtTrackingKey    = "PBS-LIMITADTRACKING"
+	ConsentKey        = "PBS-GDPRCONSENT"
+	CustomMacroPrefix = "PBS-MACRO_"
+	BidderKey         = "PBS-BIDDER"
+	IntegrationKey    = "PBS-INTEGRATION"
+)
+
+var (
+	bidLevelKeys = []string{BidIDKey, BidderKey}
+)
+
+type Provider interface {
+	// GetMacro returns the macro value for the given macro key
+	GetMacro(key string) string
+	// GetAllMacros return all the macros
+	GetAllMacros(keys []string) map[string]string
+	// SetContext set the bid and imp for the current provider
+	SetContext(bid *openrtb2.Bid, imp *openrtb2.Imp, seat string)
```

Comment:
This method implies the providers will be reused, but they are request specific. What are your plans with this?

Reply:
In that case what do you think about splitting and renaming this function into two functions such as `SetBidMacros` and `SetEventMacros`. Also, what do you think about getting rid of the reset function altogether? We can just override the values for the bid/event macros in their respective `set` functions, if those values exist and if they don't, only then we delete that Macro from the map?

Reply:
Done. Removed the SetContext function and added two separate functions  `PopulateEventMacros` and `PopulateBidMacros`